### DISCUSSION TITLE
AVIF encoding is pretty slow

### DISCUSSION THREAD

**Discussion by L3tum** [libvips/libvips#2647] [Help wanted]
Hey, this isn't really a bug report but I'm hoping that someone can help me.

We're building libvips from source (because bullseye only contains 8.10.5 as latest version) and somehow our build is *much* slower than the prebuilt Windows binaries.

Encoding the same image, with the same options, I'm looking at 13 seconds for our build and 1,5 seconds for the Windows binaries.
Ref: `--Q=50 --compression=av1 --strip=true --effort=2`

We're building libheif like this, with LIBHEIF_VERSION=1.12.0 and the libheif patch being the same that is applied to the Windows binaries, and libaom installed from Bullseye packages:

````shell
curl -fsSLO https://github.com/strukturag/libheif/releases/download/v${LIBHEIF_VERSION}/libheif-${LIBHEIF_VERSION}.tar.gz
tar zvxf libheif-${LIBHEIF_VERSION}.tar.gz
cd libheif-${LIBHEIF_VERSION}
git apply --reject --whitespace=fix ${VIPS_DIR}/patches/libheif.patch
CFLAGS=\"-g -O3 -mavx2\" CXXFLAGS=\"-g -O3 -mavx2\" CPPFLAGS=\"-g -O3 -mavx2\" ./configure --disable-go --disable-examples --disable-gdk-pixbuf --disable-rav1e --disable-dav1d --disable-libde265 --disable-x265,
make -j$(nproc)
````

and we're building libvips like this, with LIBVIPS_VERSION=8.12.2:
````shell
curl -fsSLO https://github.com/libvips/libvips/releases/download/v${LIBVIPS_VERSION}/vips-${LIBVIPS_VERSION}.tar.gz
tar zvxf vips-${LIBVIPS_VERSION}.tar.gz
cd /tmp/vips-${LIBVIPS_VERSION}
CFLAGS=\"-g -O3 -mavx2\" CXXFLAGS=\"-g -O3 -mavx2\" CPPFLAGS=\"-g -O3 -mavx2\" ./configure --with-heif=module --with-libspng=module --disable-debug --disable-doxygen --without-pdfium --disable-introspection --disable-deprecated --enable-gtk-doc-html=no --enable-gtk-doc=no
make
````

I've read that `-O3` can sometimes produce non-standard-compliant binaries, but I wouldn't expect it to be slower, especially not by that much. Is there anything else that may be amiss? Other encoders seem to perform similarly enough.

**Comment by lovell** [comment]
> libaom installed from Bullseye packages

https://packages.debian.org/bullseye/libaom-dev (v1.0.0) or https://packages.debian.org/bullseye-backports/libaom-dev (v3.2.0) ?

**Reply by L3tum** [reply]
Oh shit, I didn't even see that. That's probably the issue, lol. 
Thank you very much!
